File: packages/sdk-ruby/lib/bidstack_crm/resources/webhooks.rb
require_relative "base"

module BidstackCrm
  module Resources
    class Webhooks < Base
      def list = get("/webhooks")
      def find(webhook_id) = get("/webhooks/#{webhook_id}")
      def create(data) = post("/webhooks", body: data)
      def update(webhook_id, data) = patch("/webhooks/#{webhook_id}", body: data)
      def destroy(webhook_id) = delete("/webhooks/#{webhook_id}")
      def test(webhook_id) = post("/webhooks/#{webhook_id}/test")
    end
  end
end

File: packages/sdk-python/bidstack_crm/resources/webhooks.py
"""Webhooks resource."""
from __future__ import annotations
from typing import Any
from ._base import BaseResource


class WebhooksResource(BaseResource):
    def list(self) -> dict[str, Any]:
        return self._get("/webhooks")

    def get(self, webhook_id: str) -> dict[str, Any]:
        return self._get(f"/webhooks/{webhook_id}")

    def create(self, data: dict[str, Any]) -> dict[str, Any]:
        """Register a new webhook endpoint."""
        return self._post("/webhooks", json=data)

    def update(self, webhook_id: str, data: dict[str, Any]) -> dict[str, Any]:
        return self._patch(f"/webhooks/{webhook_id}", json=data)

    def delete(self, webhook_id: str) -> None:
        self._delete(f"/webhooks/{webhook_id}")

    def test(self, webhook_id: str) -> dict[str, Any]:
        """Fire a test ping to verify the endpoint is reachable."""
        return self._post(f"/webhooks/{webhook_id}/test")
